packages/corvid-e2e-tests/test/drivers/cliDriver.js

```diff
@@ -1,29 +1,63 @@
 const execa = require("execa");
 const which = require("npm-which")(__dirname);
-
+const { findFreePort } = require("./utils");
 const CORVID_BIN_PATH = which.sync("corvid");
 
 module.exports = ({ cwd }) => {
   const parseCommandArgs = ({ remoteDebuggingPort } = {}) =>
     remoteDebuggingPort ? `--remote-debugging-port=${remoteDebuggingPort}` : "";
 
-  const login = ({ remoteDebuggingPort } = {}) => {
-    const query = parseCommandArgs({ remoteDebuggingPort });
-    return execa.shell(`${CORVID_BIN_PATH} login ${query}`, { cwd });
+  const login = async ({ remoteDebuggingPort } = {}) => {
+    const port = remoteDebuggingPort || (await findFreePort());
+    const query = parseCommandArgs({ remoteDebuggingPort: port });
+    const command = execa.shell(`${CORVID_BIN_PATH} login ${query}`, {
+      cwd
+    });
+    return {
+      remoteDebuggingPort: port,
+      waitForCommandToEnd: () => command,
+      kill: async () => await command.kill()
+    };
+  };
+
+  const logout = async ({ remoteDebuggingPort } = {}) => {
+    const port = remoteDebuggingPort || (await findFreePort());
+    const query = parseCommandArgs({ remoteDebuggingPort: port });
+    const command = execa.shell(`${CORVID_BIN_PATH} logout ${query}`, { cwd });
+    return {
+      remoteDebuggingPort: port,
+      waitForCommandToEnd: () => command,
+      kill: async () => await command.kill()
+    };
   };
 
-  const logout = () => execa.shell(`${CORVID_BIN_PATH} logout`, { cwd });
+  const clone = async ({ editorUrl, remoteDebuggingPort } = {}) => {
```

Comment:
all commands look the same besides the command name. you can unite then all to one function and then do:

const clone = runCommand("clone")
const openEditor = runCommand("open-editor")
etc...